pkg/testframework/registry.go

```diff
@@ -482,3 +492,39 @@ func CreateEphemeralRegistry(t *testing.T, restConfig *rest.Config, namespace st
 	t.Logf("created ephemeral registry: %s (%s)", host, name)
 	return host, name, cleanup
 }
+
+func checkRoute(host string) error {
+	tr := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	}
+
+	url := fmt.Sprintf("https://%s/v2/", host)
+	client := &http.Client{Transport: tr}
+	res, err := client.Get(url)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK && res.StatusCode != http.StatusUnauthorized {
```

Comment:
nit - add a comment that `http.StatusUnauthorized` indicates that the registry is operating normally.

Reply:
Good call. Done.